### PR TITLE
[srgssr] rework to use API v2 and HTTPS

### PR DESCRIPTION
**Pull request by vszakats**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

* use v2 API endpoint
* use HTTPS protocol for metadata downloads
* prefer HTTPS streams
* add HTTPS test
* remove a test with URL no longer availabe
  (id `677f5829-e473-4823-ac83-a1087fe97faa`)
* fix test issues (e.g. id `6348260`)
* v2 API no longer offers `publishedDate`, only a `date` which may be slightly different. Tests updated accordingly.

### CODE REVIEW

**Comment by vszakats** [comment]
Ref: https://github.com/ytdl-org/youtube-dl/pull/24558

**Comment by vszakats** [comment]
@dstftw What do you think about this PR? Any of concern or something to fix?

File: youtube_dl/extractor/srgssr.py [hunk 1]
```diff
@@ -56,42 +67,50 @@ def _real_extract(self, url):
...
-        title = metadata['title']
-        description = metadata.get('description')
-        created_date = media_data.get('createdDate') or metadata.get('createdDate')
+        chapter = media_data.get('_chapter', {})
+        if media_data.get('_segment'):
+            metadata = media_data['_segment']
+        else:
+            metadata = chapter
+
+        title = metadata.get('title')
```
Comment (dstftw) [inline youtube_dl/extractor/srgssr.py:76]: Read coding conventions on mandatory metadata.

File: youtube_dl/extractor/srgssr.py [hunk 2]
```diff
@@ -56,42 +67,50 @@ def _real_extract(self, url):
...
+        description = metadata.get('description', '')
```
Comment (dstftw) [inline youtube_dl/extractor/srgssr.py:77]: Lack of data is denoted by None not empty string.

**Comment by vszakats** [comment]
So I'm guessing this API doesn't fit your strict criteria. Let's leave it insecure and using the obsolete API.

**Pull request closed without merging**